test/unittest/blas3/blas3_trsm_test.cpp

```diff
@@ -0,0 +1,95 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas3_gemm_test.cpp
+ *
+ **************************************************************************/
+
+#include "blas_test.hpp"
+
+template <typename scalar_t>
+using combination_t = std::tuple<int, int, char, char, char, char>;
+
+template <typename scalar_t>
+void run_test(const combination_t<scalar_t> combi) {
+  int m;
+  int n;
+  char transA;
+  char side;
+  char diag;
+  char triangle;
+  std::tie(m, n, transA, side, diag, triangle) = combi;
+
+  using data_t = utils::data_storage_t<scalar_t>;
+
+  scalar_t alpha = scalar_t{1};
+  const int lda = side == 'l' ? m : n;
+  const int ldb = m;
+  const int k = side == 'l' ? m : n;
+
+  const int sizeA = k * lda;
+  const int sizeB = n * ldb;
+
+  std::vector<scalar_t> A(sizeA);
+  std::vector<scalar_t> B(sizeB);
+
+  const scalar_t diagValue =
+      diag == 'u' ? scalar_t{1} : random_scalar(scalar_t{1}, scalar_t{10});
+
+  fill_triangle(A, k, k, lda, triangle, diagValue);
+  fill_random(B);
```

Comment:
This is one of the examples that you need all random function to use the same seed and algorithm to generate so that it will be reproducible and predictable, So in this case B and A will use the same random seed to generate the matrix

Reply:
The internals of the fill triangle and fill random were updated to use the same random generator engine